src/main/java/com/dataguise/saas/service/DetectionTaskService.java

```diff
@@ -0,0 +1,192 @@
+
+            for (Integer connectionId : connectionIdList) {
+                builder.addConnection(connectionId);
+            }
+            logger.debug("Detection task builder bean to driver: "+builder);
+            discoverTask = builder.build().dgDiscoverTask;
+        } catch (DgDrvException e) {
+            logger.info(e.getMessage());
+            throw new BadGatewayException("Driver unable to construct detection task.", HttpStatus.BAD_GATEWAY);
+        }
+        return discoverTask;
+    }
+
+    /**
+     * Create LFA-S3 detection task
+     * @param lfaDetectionTask
+     * @return
+     * @throws BadGatewayException
+     */
+    public Integer createLFATask(@RequestBody CloudTaskBean lfaDetectionTask) throws BadGatewayException {
+        Integer taskId = null;
+        try {
+            taskId = restDriver.createLFADetectionTask(lfaDetectionTask);
+        } catch (DgDrvException e) {
+            throw new BadGatewayException(e.getMessage(), HttpStatus.BAD_GATEWAY);
+        }
+        logger.info("LFA Detection task created with id: " + taskId);
+        return taskId;
+    }
+
+    /**
+     * Construct the CloudTaskBean to create LFA-S3 detection task.
+     * @param scanLocations
+     * @return
+     * @throws BadGatewayException
+     * @throws InternalServerException
+     */
+    public CloudTaskBean constructLFADetectionTask(List<String> scanLocations) throws BadGatewayException,InternalServerException {
+        CloudTaskBean lfaDiscoverTask = null;
+        List<String> excludeList = new ArrayList<>();
+        try {
+            String taskDetail = taskBuilderService.constructTaskName(taskBuilderService.DETECTION_TYPE);
+            LFADetectionTask.LFADetectionTaskBuilder builder = new LFADetectionTask.LFADetectionTaskBuilder(taskDetail, taskDetail)
+                    .setTaskType(LFADetectionTask.TaskTypes.Discover)
+                    .setBatchSize(30)  //Default batch size
+                    .setClusterProvisionType(LFADetectionTask.ClusterProvisionType.Manual)
+                    .setDeleteOriginal(false)
+                    .setFullFileScan(false)
+                    .setDiscoveryCriteria(false)
+                    .setIncrementalDiscovery(true)
+                    .setSkipped(true)
+                    .setStructured(false)
+                    .setVerifyStructure(false)
+                    .setScanLocations(scanLocations)
+                    .setClusterType(LFADetectionTask.ClusterTypes.UnixFiles)
+                    .setExcludedFileExtensions(excludeList)
+                    .setExcludedScanPathList(excludeList)
+                    .setSamplingId(15)  //Sampling Id=15 for moduleType='file'(Top 1000 rows)
+                    .setOutputColumnForm("replace")
+                    .setValueSeparator(",");
+
+            builder = taskBuilderService.addTaskBuilderPolicies(builder);
+
+            logger.debug("LFA Detection task builder bean to driver: "+builder);
```

Comment:
**Reporter**: PMD    
    **Rule**: GuardLogStatement
    **Severity**: ERROR
    **File**: /work/src/main/java/com/dataguise/saas/service/DetectionTaskService.java L183
    
    Logger calls should be surrounded by log level guards.&#10;&#10;&#10;Best Practices https://pmd.github.io/pmd/pmd_rules_java_bestpractices.html#guardlogstatement
    
*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a1563054335>*